### PR TITLE
Some tests and cleanup for glueings.

### PR DESCRIPTION
**Pull request by HechtiDerLachs**

### CODE REVIEW

File: experimental/Schemes/SpecOpen.jl [hunk 1]
```diff
@@ -711,13 +711,13 @@ function Base.show(io::IO, f::SpecOpenMor)
   #given by the rational map $(generic_fractions(f))")
 end
 
-function SpecOpenMor(U::T, V::T, f::Vector{<:RingElem}; check::Bool=true) where {T<:SpecOpen}
+function SpecOpenMor(U::SpecOpen, V::SpecOpen, f::Vector{<:RingElem}; check::Bool=true)
```
Comment (simonbrandhorst) [inline experimental/Schemes/SpecOpen.jl:714]: Why change this (here and elsewhere) ? Do you want to allow two `SpecOpen` of a different type?

File: experimental/Schemes/SpecOpen.jl [hunk 2]
```diff
@@ -784,7 +784,9 @@ function restrict(
   end
   inc = inclusion_morphism(U, domain(f), check=check)
   help_map = compose(inc, f, check=check)
-  return SpecOpenMor(U, V, maps_on_patches(help_map), check=check)
+  Y = ambient(V)
+  h = [restrict(g, domain(g), Y, check=check) for g in maps_on_patches(help_map)]
```
Comment (simonbrandhorst) [inline experimental/Schemes/SpecOpen.jl:788]: So `restrict` can also extend the codomain?

File: experimental/Schemes/Types.jl [hunk 3]
```diff
@@ -259,10 +259,19 @@ the list ``f₁,…,fᵣ`` as the *generators* for ``U``.
       end
     end
     U = new{SpecType, typeof(base_ring(X))}(X, f)
-    U.intersections = Dict{Tuple{Int, Int}, SpecType}()
+    U.intersections = Dict{Tuple{Int, Int}, AbsSpec}()
```
Comment (simonbrandhorst) [inline experimental/Schemes/Types.jl:262]: Why this change?

File: experimental/Schemes/SpecOpen.jl (hunk 1 above)
Comment (HechtiDerLachs) [reply experimental/Schemes/SpecOpen.jl:714]: Yes, exactly. We need that flexibility.

File: experimental/Schemes/SpecOpen.jl (hunk 2 above)
Comment (HechtiDerLachs) [reply experimental/Schemes/SpecOpen.jl:788]: No. But the maps on the patches of a `SpecOpenMor` are required to take values in the `ambient` of the codomain. And that needs to be assure when restricting.

File: experimental/Schemes/Types.jl (hunk 3 above)
Comment (HechtiDerLachs) [reply experimental/Schemes/Types.jl:262]: Again, because we need the flexibility and this was a deprecated leftover from older code.

File: experimental/Schemes/SpecOpen.jl (hunk 2 above)
Comment (simonbrandhorst) [reply experimental/Schemes/SpecOpen.jl:788]: So it is kind of a hack?

**Review by simonbrandhorst — APPROVED** [review]

File: experimental/Schemes/SpecOpen.jl (hunk 2 above)
Comment (HechtiDerLachs) [reply experimental/Schemes/SpecOpen.jl:788]: I wouldn't say so, really...

It is basically just how I constructed it for now. For any `SpecOpen` `U` which is covered by affine patches `U_i`, a  `SpecOpenMor` `f : U -> V \subset Y` is given by `SpecMor`s `f_i : U_i -> Y` which glue on the overlaps `U_i \cap U_j` and whose image is contained in `V`. 

It would just be way too restrictive to require `V` to be partitioned in a compatible way so that the `f_i` would take values in affine patches of `V`. So the requirement is that the codomain of `f_i` must be `Y`. Now when we restrict to closed subsets `A` in `U` and `B` in `V`, then the ambient scheme of `B` can not be `Y` anymore and we have to restrict the map on the affine patches accordingly.

**Comment by HechtiDerLachs** [comment]
We should probably be a bit careful with the automerge. I just saw that, due to the bug in the documentation, our other PR is also still not merged. I might have to rebase on that, eventually.

**Comment by simonbrandhorst** [comment]
Alright, I will turn off the automerge for now.
